src/Knoop/Node/src/insert.hpp

```diff
@@ -1,7 +1,23 @@
-template< typename T >
-Node& insert( std::string str, T&& t ){
+template< typename Str, typename T,
+          typename = void_t< decltype( std::declval< map_type >()
+                                       .at( std::declval< Str >() ) ) > >
+Node& insert( Str&& str, T&& t ){
   auto& map = std::experimental::get< map_type >( core );
-  map[ str ].push_back( std::make_shared< Node >( t ) );
+  const auto p = map.emplace( str, std::forward<T>(t) );
+  if ( not p.second ){ throw std::runtime_error("didn't work"); }
```

Comment:
Do we want to include `njoy::Log` here and log this message?

Reply:
The message should be better, but logging is not appropriate in all circumstances. Let's delegat that to the client